libs/@local/codegen/src/definitions/fields.rs

```diff
@@ -0,0 +1,66 @@
+use alloc::borrow::Cow;
+
+use specta::datatype;
+
+use super::Type;
+
+#[derive(Debug, Clone)]
+pub struct Field {
+    // TODO: Export doc strings in codegen
+    //   see https://linear.app/hash/issue/H-4473/export-doc-strings-in-codegen
+    pub r#type: Type,
+}
+
+impl Field {
+    pub(crate) fn from_specta(
+        field: &datatype::Field,
+        type_collection: &specta::TypeCollection,
+    ) -> Self {
+        Self {
+            r#type: Type::from_specta(
+                field.ty().unwrap_or_else(|| {
+                    todo!("https://linear.app/hash/issue/H-4472/allow-field-skipping-in-codegen")
+                }),
+                type_collection,
+            ),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum Fields {
+    Named {
+        fields: Vec<(Cow<'static, str>, Field)>,
+    },
+    Unnamed {
+        fields: Vec<Field>,
+    },
+    Unit,
+}
```

Comment:
non-blocking, it might make sense to instead replicate how rustc handles and names this:

https://github.com/rust-lang/rust/blob/efcbb94dccbe0fcc1076f031f88332db4a6a2a52/compiler/rustc_hir/src/hir.rs#L3877-L3896

https://github.com/rust-lang/rust/blob/efcbb94dccbe0fcc1076f031f88332db4a6a2a52/compiler/rustc_hir/src/hir.rs#L3979-L3993

https://github.com/rust-lang/rust/blob/efcbb94dccbe0fcc1076f031f88332db4a6a2a52/compiler/rustc_hir/src/hir.rs#L4294-L4297

Reply:
I think renaming `Field` makes sense, but I would incorporate this in a follow-up for simplicity (as this does not really hurt to have now).